src/main/kotlin/org/codefreak/codefreak/graphql/api/AutograderApi.kt

```diff
@@ -0,0 +1,386 @@
+package org.codefreak.codefreak.graphql.api
+
+import com.expediagroup.graphql.annotations.GraphQLID
+import com.expediagroup.graphql.annotations.GraphQLIgnore
+import com.expediagroup.graphql.annotations.GraphQLName
+import com.expediagroup.graphql.spring.operations.Mutation
+import com.expediagroup.graphql.spring.operations.Query
+import java.util.UUID
+import org.codefreak.codefreak.auth.Authority
+import org.codefreak.codefreak.entity.Answer
+import org.codefreak.codefreak.entity.Assignment
+import org.codefreak.codefreak.entity.Grade
+import org.codefreak.codefreak.entity.GradeDefinition
+import org.codefreak.codefreak.entity.PointsOfEvaluationStep
+import org.codefreak.codefreak.entity.Submission
+import org.codefreak.codefreak.entity.Task
+import org.codefreak.codefreak.entity.UserAlias
+import org.codefreak.codefreak.graphql.BaseDto
+import org.codefreak.codefreak.graphql.BaseResolver
+import org.codefreak.codefreak.graphql.ResolverContext
+import org.codefreak.codefreak.service.AnswerService
+import org.codefreak.codefreak.service.AssignmentService
+import org.codefreak.codefreak.service.SubmissionService
+import org.codefreak.codefreak.service.UserAliasService
+import org.codefreak.codefreak.service.evaluation.EvaluationService
+import org.codefreak.codefreak.service.evaluation.GradeDefinitionService
+import org.codefreak.codefreak.service.evaluation.GradeService
+import org.codefreak.codefreak.service.evaluation.PointsOfEvaluationStepService
+import org.slf4j.LoggerFactory
+import org.springframework.security.access.annotation.Secured
+import org.springframework.stereotype.Component
+
+/**
+ * GradeDefinition for query
+ */
+@GraphQLName("GradeDefinition")
+class GradeDefinitionDto(@GraphQLIgnore val entity: GradeDefinition, ctx: ResolverContext) : BaseDto(ctx) {
+
+  @GraphQLID
+  val id = entity.id
+  val active = entity.active
+  val maxPoints = entity.maxPoints
+  val minorError = entity.minorError
+  val majorError = entity.majorError
+  val criticalError = entity.criticalError
+}
+
+/**
+ * GradeDefinitionMaxDto for query. Main purpose
+ */
+@GraphQLName("GradeDefinitionMax")
+class GradeDefinitionMaxDto(@GraphQLIgnore val entity: GradeDefinition, ctx: ResolverContext) : BaseDto(ctx) {
+
+  @GraphQLID
+  val id = entity.id
+  val maxPoints = entity.maxPoints
+  val active = entity.active
+}
+
+/**
+ * PointsOfEvaluation Dto for query
+ */
+@GraphQLName("PointsOfEvaluationStep")
+class PointsOfEvaluationStepDto(@GraphQLIgnore val entity: PointsOfEvaluationStep?, ctx: ResolverContext) : BaseDto(ctx) {
```

Comment:
If there is no Autograding there wont be a PointsOfEvaluationStep entity.

Reply:
yeah but  then, why create a DTO at all? in that case the DTO may as well be null

Reply:
The DTO is null if there is no PointsOfEvaluationStep present, yes. This is the case if the Evaluation is running or no autograding is active. The EvaluationStepPanel.tsx contains a PointsEdit.tsx Field. If the DTO returns null, this field wont be shown. Otherwise Grading is active and it will be shown as simple view for students or edit field for teachers.
Same approach was done on the GradeDto.